frontend/main/app/AppController.js

```diff
@@ -62,15 +78,22 @@ export class AppController {
     * @param {Object} params - Additional parameters to pass to the view.
     */
    async navigate(viewName, params = {}) {
-     if (!this.#views[viewName]) {
-       throw new Error(`View "${viewName}" not found.`);
-     }
-      this.#currentView = this.#views[viewName];
-     if(viewName === 'productPage'){
-      this.render(params.prodid);
-     }else{
-      this.render();
+    if (!this.#views[viewName]) {
+      throw new Error(`View "${viewName}" not found.`);
+    }
+
+     // Restrict access to certain views based on authentication
+    if(!authService.isLoggedIn() && (viewName !== 'auth' || 'marketplace' )){
```

Comment:
|| 'marketplace' doesn't do anything here, 'marketplace' will always evaluate to true.